src/websocket.c

```diff
@@ -355,40 +361,42 @@ int send_digest_base64(int socket_fd, const unsigned char* message_digest) {
  *            at least 25 chars long.
  * @returns 0 on success, error code otherwise.
  */
-int read_websocket_header(int socket_fd, unsigned int skip_bytes,
+int read_websocket_header(Connection* conn, unsigned int skip_bytes,
                           char* expected_protocol, char* key) {
   // Neither of these limits is in the HTTP spec, but they are the limits
   // enforced by the Apache defaults. In particular, it is recommended that
   // there not be more than 8K of headers total, so these limits are generous.
   // http://stackoverflow.com/questions/686217/maximum-on-http-header-values
-  const static unsigned int MAX_HEADER_COUNT = 1024;
-  const static unsigned int MAX_HEADER_LENGTH = 8192;
+  // They are #define'd because declaring arrays of variable size, even if the
+  // variable is const and static, is a gcc extension and not part of C.
+  #define MAX_HEADER_COUNT 1024
+  #define MAX_HEADER_LENGTH 8192
```

Comment:
Optional:
* `MAX_HEADER_COUNT` can still be an int
* To me it's less gross to just move 383 into line 373's spot as `line[8192]` and just do `sizeof` to get the value back when needed. I think it's actually better to do it that way even if you don't get rid of `MAX_HEADER_LENGTH`. e.g.:

```
ws_readline(conn, line, sizeof(line))
```